# Dan Miller: Exploratory Data Analysis Project
## This Notebook will be used for Project 6 of Data Analytics Fundamentals
## 2/14/2025

## 1. Imports
1. pandas
2. seaborn
3. matplotlib

In [19]:
import pandas as pd
import seaborn as sns
import matplotlib

## 2. Load Data

Loading the Tips dataset using sns.load_dataset().

This dataset is one of the seaborn datasets that are easy to work with and require little hassle to load.

In [20]:
#Load the Tips dataset into pandas DataFrame
tips_df: pd.DataFrame = sns.load_dataset('tips')

#List column names
tips_df.columns

#Inspect first few rows of the DataFrame
tips_df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

## 3. Initial Data Inspection

In [21]:
#Specify the number of rows to display
tips_df.head(10)

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4
5       25.29  4.71    Male     No  Sun  Dinner     4
6        8.77  2.00    Male     No  Sun  Dinner     2
7       26.88  3.12    Male     No  Sun  Dinner     4
8       15.04  1.96    Male     No  Sun  Dinner     2
9       14.78  3.23    Male     No  Sun  Dinner     2

In [22]:
#Check the shape of the DataFrame
tips_df.shape

(244, 7)

In [23]:
#Display the data types of the columns
tips_df.dtypes

total_bill     float64
tip            float64
sex           category
smoker        category
day           category
time          category
size             int64
dtype: object

This dataset has 7 features in total, 4 of which are categorical while the other 3 are numerical.  In total, there are 244 individual rows, which is a large enough size to create useful statistics from.

## 4. Initial Descriptive Statistics